run_tests.py

```diff
@@ -0,0 +1,176 @@
+#!/bin/sh
+"exec" "python3" "-u" "-B" "$0" "$@"
+# Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+# other BLT Project Developers. See the top-level LICENSE file for details
+# 
+# SPDX-License-Identifier: (BSD-3-Clause)
+
+import os
+import sys
+import subprocess
+import glob
+import re
+import argparse
+import string
+
+from functools import partial
+
+# Since we use subprocesses, flushing prints allows us to keep logs in
+# order.
+print = partial(print, flush=True)
+
+def sexe(cmd):
+    """ Helper for executing shell commands. """
+    p = subprocess.Popen(cmd,
+                            shell=True,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT)
+    out = p.communicate()[0]
+    out = out.decode('utf8')
+    return p.returncode, out
+
+def cmake_build_project(path_to_test: string, path_to_blt: string, host_config: string, is_base: bool):
+    base_or_downstream = "base" if is_base else "downstream"
+    install_flag = "-DCMAKE_INSTALL_PREFIX" if is_base else "-Dbase_install_dir"
+    blt_path_flag = "-DPATH_TO_BLT"

```

Comment:
Just for consistency because it is so widely used, please change the name of this variable to `BLT_SOURCE_DIR`